sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/ClassicTableTypeMapping.java

```diff
@@ -61,22 +57,23 @@ public ClassicTableTypeMapping() {
     hiveToClientMap.put(TableType.MATERIALIZED_VIEW.toString(),
             ClassicTableTypes.MATERIALIZED_VIEW.toString());
 
-    clientToHiveMap.putAll(ClassicTableTypes.TABLE.name(), Arrays.asList(
-        TableType.MANAGED_TABLE.name(), TableType.EXTERNAL_TABLE.name()));
-    clientToHiveMap.put(ClassicTableTypes.VIEW.name(), TableType.VIRTUAL_VIEW.name());
+    clientToHiveMap.put(ClassicTableTypes.TABLE.name(),
+        List.of(TableType.MANAGED_TABLE.name(), TableType.EXTERNAL_TABLE.name()));
+    clientToHiveMap.put(ClassicTableTypes.VIEW.name(),
+        List.of(TableType.VIRTUAL_VIEW.name()));
     clientToHiveMap.put(ClassicTableTypes.MATERIALIZED_VIEW.toString(),
-            TableType.MATERIALIZED_VIEW.toString());
+        List.of(TableType.MATERIALIZED_VIEW.toString()));
   }
 
   @Override
   public String[] mapToHiveType(String clientTypeName) {
-    Collection<String> hiveTableType = clientToHiveMap.get(clientTypeName.toUpperCase());
+    List<String> hiveTableType = clientToHiveMap.get(clientTypeName.toUpperCase());
     if (hiveTableType == null) {
       LOG.warn("Not supported client table type {}",
         MDC.of(LogKeys.TABLE_TYPE, clientTypeName));
       return new String[] {clientTypeName};
     }
-    return Iterables.toArray(hiveTableType, String.class);
+    return hiveTableType.toArray(new String[0]);
```

Comment:
nit: `new String[0]` might as well exist as a constant,  there should likely be other `toArray` calls that require this constant, and of course, this can be done in a separate pr